src/main/kotlin/org/cqfn/save/adapter/Adapter.kt

```diff
@@ -0,0 +1,49 @@
+package org.cqfn.save.adapter
+
+import io.github.detekt.sarif4k.Result
+import io.github.detekt.sarif4k.Run
+import io.github.detekt.sarif4k.SarifSchema210
+import io.github.detekt.sarif4k.Tool
+import io.github.detekt.sarif4k.Version
+import kotlinx.serialization.StringFormat
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import java.io.InputStreamReader
+import java.io.OutputStreamWriter
+import kotlin.reflect.KType
+
+abstract class Adapter<T : Any>(
```

Comment:
extract core logic please

Reply:
Split into `Adapter`, `KxSerializationAdapter`, `SarifAdapter` and created a factory method